girder/api/v1/notification.py

```diff
@@ -105,3 +106,28 @@ def streamGen():
 
                 time.sleep(wait)
         return streamGen
+
+    @access.cookie
+    @access.token
+    @autoDescribeRoute(
+        Description('List notification events')
+        .notes('This endpoint can be used for manual long-polling when '
+               'SSE support is disabled or otherwise unavailable.')
+        .param('since', 'Filter out events before this time stamp.',
+               dataType='integer', required=False)
+        .errorResponse()
+        .errorResponse('You are not logged in.', 403)
+    )
+    def listNotifications(self, since):
+        user, token = self.getCurrentUser(returnToken=True)
+        timestamp = time.time()
+
+        if since is not None:
+            since = datetime.utcfromtimestamp(since)
+        notifications = list(NotificationModel().get(user, since, token=token))
+
+        if notifications:
+            timestamp = max(n['updatedTime'] for n in notifications)
```

Comment:
We could just control this completely on the client side without the need to send any headers at all. The client can simply poll this endpoint, and if it receives nothing in response, it just polls with the same parameters as before. If it receives a non-empty list in response, it iterates through the list and pulls the max timestamp from it, and uses that on the next poll.

If we choose to do it that way, we may want to support ISO 8601 dates instead of (or in addition to) epoch times for the `since` parameter.

Reply:
@zachmullen Yup, I'd be fine if we want to implement this on the client-side instead. As I mentioned before:
> Alternatively, do each of the included events include a timestamp? Would an application be able to use the latest timestamp of all events as the since value?

Reply:
@jbeezley @zachmullen Also, the current use of Unix times as the value of the `Date` HTTP header is not compliant with the RFC, the value must be encoded as an RFC-1123 date.

Reply:
I'll put in a follow up PR shortly that will simplify this endpoint.

Reply:
refs #2769